store/tikv/2pc.go

```diff
@@ -85,6 +85,11 @@ type twoPhaseCommitter struct {
 	detail        *execdetails.CommitDetails
 }
 
+type mutationEx struct {
+	pb.Mutation
+	hasContract bool
```

Comment:
I think `hasContract` is redundant, because Mutation has member `precondition`.

Reply:
If this field is removed, what to do when there is a `precondition` error? 
We have to distinguish the contract that must hold, and some precondition that could fail.

Reply:
Done